FiniteStateMachine/src/statemachine/State.java

```diff
@@ -2,11 +2,9 @@
 
 import org.jetbrains.annotations.NotNull;
 
-import java.util.EnumSet;
+import java.util.Set;
 
 
-public interface State<E extends Enum<E> & State<E>> {
```

Comment:
I'm still trying to wrap my head around an example where the states aren't represented by and Enum or a proxy for one.